packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -51,13 +52,13 @@ export const satisfies = (zcf, seat, update) => {
 /** @type {Swap} */
 export const swap = (zcf, leftSeat, rightSeat) => {
   try {
-    rightSeat.decrementBy(harden(leftSeat.getProposal().want));
-    leftSeat.incrementBy(harden(leftSeat.getProposal().want));
-
-    leftSeat.decrementBy(harden(rightSeat.getProposal().want));
-    rightSeat.incrementBy(harden(rightSeat.getProposal().want));
-
-    zcf.reallocate(leftSeat, rightSeat);
+    atomicTransfer(
+      zcf,
+      harden([
+        [rightSeat, leftSeat, leftSeat.getProposal().want],
+        [leftSeat, rightSeat, rightSeat.getProposal().want],
```

Comment:
These are different amounts and keywords

Reply:
I don't understand. AFAICT, the new code should be exactly equivalent to the old. What am I missing?

Reply:
I now understand the similar issue you've raised elsewhere. However, I don't think it applies here.

Each line only has one AmountsKeywordRecord, and thus serves as both the fromAmounts and twoAmounts for the oneway transfer represented by that line alone. So each line is balanced by itself. Like the original, this is doing a pair of one-way transfers.

Reply:
Resolving